website/search/elastic_search.py

```diff
@@ -489,7 +495,7 @@ def update_file(file_, index=None, delete=False):
     index = index or INDEX
 
     # TODO: Can remove 'not file_.name' if we remove all base file nodes with name=None
-    if not file_.name or not file_.node.is_public or delete or file_.node.is_deleted or file_.node.archiving:
+    if not file_.name or not file_.node.is_public or delete or file_.node.is_deleted or file_.node.archiving or 'spam_confirmed' in file_.node.creator.system_tags:
```

Comment:
This is adding a lot of joins (on `node`, `creator`, and `tags`) every time we update a file in ES. For now, it might make more sense to just delete the files in the `update_user` function:

```python
def update_user(..):

    ...
    for quickfile in BaseFileNode.objects.filter(...).only('_id'):
         client().delete(
             index=index,
             doc_type='file',
             id=file_._id,
             refresh=True,
             ignore=[404]
        )
```